r/work.R

```diff
@@ -41,7 +41,33 @@ run_post <- function(req, post_fun, data) {
     debug_step <- Sys.getenv("DEBUG_STEP", unset = "")
 
     handle_debug(req, debug_step)
-    post_fun(req, data)
+
+    message(rep("✧",100))
+    message("➥ Starting ",sub("run","",basename(req$path)))
+    message("Input:")
+    message(str(req$body))
+
+    tryCatch({
+            message("\nSeurat logs:")
+            message("➡️ \n")
+            tstart <- Sys.time()
+            res <- post_fun(req, data)
+            message("\n⬅️")
+
+            message("\nResult head: ")
+            message(str(head(res,10)))
+
+            ttask <- format(Sys.time()-tstart, digits = 2)
+            message("\n⏱️ Time to complete ", req$body$name, " for experiment ", experiment_id, ": ", ttask, '\n')
+            message("✅ Finished ", req$body$name)
+            message(rep("✧",100))
+            return(res)
+        },
+        error = function(e) {
+            flog.error("🚩 --------- 🚩")
+            message("Error at worker task: ", e)
```

Comment:
in the pipeline the error is printed with `e$message` instead of just e, do you know if it matters? or do we see all the information anyways?

Reply:
Modified! Looks better